### PR TITLE
ci: add release & publish workflows

### PR DESCRIPTION
**Pull request by yshrsmz**

### CODE REVIEW

File: bucketeer/gradle.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-POM_ARTIFACT_ID=sdk-android
+POM_ARTIFACT_ID=android-client-sdk
```
Comment (yshrsmz) [inline bucketeer/gradle.properties:1]: module name is `io.bucketeer:android-client-sdk`

File: gradle.properties [hunk 2]
```diff
@@ -5,13 +5,10 @@ android.minSdkVersion=21
...
-publish.groupId=jp.bucketeer
-publish.artifactId=sdk-android
-publish.vcsUrl=https://github.com/ca-dp/bucketeer-android-sdk
-#
 # com.vanniktech.maven.publish
-GROUP=jp.bucketeer
+GROUP=io.bucketeer
+# this VERSION_NAME is just a placeholder
+# actual VERSION_NAME is provided by CI
 VERSION_NAME=0.1.0-SNAPSHOT
```
Comment (yshrsmz) [inline gradle.properties:12]: In publish-maven workflow, this value should be overridden by `ORG_GRADLE_PROJECT_VERSION_NAME` environment variable.

File: .github/workflows/publish-maven.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: publish-maven
```
Comment (yshrsmz) [inline .github/workflows/publish-maven.yml:1]: I confirmed that this workflow can upload artifacts to the sonatype staging repository.
(though I don't know if it actually `closeAndReleaseRepository`)

https://github.com/yshrsmz/android-client-sdk/actions/runs/3140792650/jobs/5102548182

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+              {"type":"build","section":"Build System","hidden":false},
+              {"type":"chore","section":"Miscellaneous","hidden":false},
+              {"type":"feat","section":"Features","hidden":false},
+              {"type":"fix","section":"Bug Fixes","hidden":false},
+              {"type":"perf","section":"Performance Improvements","hidden":false},
+              {"type":"revert","section":"Reverts","hidden":false}
+            ]
+          release-type: simple
+          bump-minor-pre-major: true
+          release-as: 2.0.0
```
Comment (yshrsmz) [inline .github/workflows/release.yml:27]: set initial release version as 2.0.0.
This should be removed after the initial release.

**Review by cre8ivejp — APPROVED** [review]
Awesome!!
Thank you!

**Comment by yshrsmz** [comment]
thanks

**Pull request merged**